File: examples/java/operations/invoke/invoke-with-config.java
import software.amazon.lambda.durable.DurableContext;
import software.amazon.lambda.durable.DurableHandler;
import software.amazon.lambda.durable.config.InvokeConfig;

public class InvokeWithConfig extends DurableHandler<OrderEvent, OrderResult> {

    @Override
    public OrderResult handle(OrderEvent event, DurableContext context) {
        InvokeConfig config = InvokeConfig.builder()
            .tenantId(event.getTenantId())
            .build();

        return context.invoke(
            "process-order",
            "order-processor-function:live",
            event,
            OrderResult.class,
            config
        );
    }
}

File: examples/csharp/operations/invoke/invoke-with-config.cs
using Amazon.Lambda.Core;
using Amazon.Lambda.DurableExecution;

public class InvokeWithConfigExample
{
    public Task<DurableExecutionInvocationOutput> Handler(
        DurableExecutionInvocationInput input, ILambdaContext context)
        => DurableFunction.WrapAsync<OrderEvent, OrderResult>(Workflow, input, context);

    private async Task<OrderResult> Workflow(OrderEvent input, IDurableContext ctx)
    {
        var config = new InvokeConfig
        {
            TenantId = input.TenantId,
        };

        OrderResult result = await ctx.InvokeAsync<OrderEvent, OrderResult>(
            "order-processor-function:live",
            input,
            name: "process-order",
            config: config);

        return result;
    }
}

public record OrderEvent(string OrderId, string TenantId);
public record OrderResult(string Status);
